src/graph/planner/match/MatchClausePlanner.cpp

```diff
@@ -108,32 +116,54 @@ StatusOr<SubPlan> MatchClausePlanner::transform(CypherClauseContextBase* clauseC
   }
 
   auto* matchClauseCtx = static_cast<MatchClauseContext*>(clauseCtx);
-  auto& nodeInfos = matchClauseCtx->nodeInfos;
-  auto& edgeInfos = matchClauseCtx->edgeInfos;
   SubPlan matchClausePlan;
-  size_t startIndex = 0;
-  bool startFromEdge = false;
-
-  NG_RETURN_IF_ERROR(findStarts(matchClauseCtx, startFromEdge, startIndex, matchClausePlan));
-  NG_RETURN_IF_ERROR(
-      expand(nodeInfos, edgeInfos, matchClauseCtx, startFromEdge, startIndex, matchClausePlan));
+  // All nodes ever seen in current match clause
+  std::unordered_set<std::string> nodeAliasesSeen;
+  // TODO: Maybe it is better to rebuild the graph and find all connected components.
+  auto& pathInfos = matchClauseCtx->paths;
+  for (auto iter = pathInfos.begin(); iter < pathInfos.end(); ++iter) {
```

Comment:
iter != pathInfos.end()

Reply:
Emmm, seems not that much necessary, the std::vector::iterator supports both < and !=.